src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -387,6 +387,11 @@ def to(self, *args, **kwargs):
 
         device = device or device_arg
 
+        def model_has_device_map(model):
```

Comment:
@DN6 it would make sense to make this a separate utility instead of having redefine three times. WDYT?

Reply:
Yup, you can add as a util function inside pipeline_utils.